### ISSUE TITLE
(feature) add flag to [sysexin] to make it return lists

### ISSUE THREAD

**Issue by mariobuoninfante** [pure-data/pure-data#1067]
Currently when dealing with incoming sysex messages, you have to parse them byte by byte. Totally doable of course, but I was wondering if we could have a flag (ie [sysexin -l]) to make the object return a list.
In this way we can get rid of 12 objects.

<img width="272" alt="Screen Shot 2020-06-11 at 14 47 15" src="https://user-images.githubusercontent.com/20879405/84392663-8f219700-abf2-11ea-9dd7-114f8d291543.png">

**Comment by Jonathhhan** [comment]
You can also do it like that:
![sysex](https://user-images.githubusercontent.com/41275844/84399059-f1819400-ac00-11ea-88b0-61f5d51f5b31.PNG)
But I like the idea with the list argument.

**Comment by mariobuoninfante** [comment]
absolutely, there are different solutions, but still I suppose there would be some convenience in having the object itself dealing with that.
Also, this would match the behavior of [midiout] that since 0.48 (if I'm not wrong) accepts lists as well as single bytes.

<img width="173" alt="Screen Shot 2020-06-11 at 16 26 03" src="https://user-images.githubusercontent.com/20879405/84405590-41ac2680-ac00-11ea-9aff-527adf4d4738.png">

**Comment by Jonathhhan** [comment]
Perhaps it is not even necessary to parse them byte by byte? I always needed sysex as a list.

**Comment by mariobuoninfante** [comment]
well, that would be a breaking change that I suppose is not really needed.
Also there could be situations where you want to get the msg byte by byte. In the end, MIDI is a serial connection, so 1 byte at the time.

**Comment by porres** [comment]
I know the thread shouldn't be about the best way to patch this, but I think this is the most elegant one

<img width="241" alt="Screen Shot 2020-06-14 at 17 40 38" src="https://user-images.githubusercontent.com/5360142/84603640-42de8d00-ae66-11ea-8f1c-71da928518bd.png">

**Comment by mariobuoninfante** [comment]
well, I have to say it is a good solution indeed.
But, I'm a "clean the list/array before using it" kind of person,
Probably this is not needed in this scenario since I don't think there is the possibility that [sysexin] skips a byte (ie malformed/incomplete sysex) - but I haven't double checked tbh.

**Comment by Jonathhhan** [comment]
also off-topic:
i used [textfile] because it is the fastest solution for very large lists. with smaller lists the difference does matter less (or not at all). with all methods you can also clear before using them.
![sysex2](https://user-images.githubusercontent.com/41275844/84605802-80600c00-aea0-11ea-8903-c3bc44cc5f50.PNG)

**Comment by Spacechild1** [comment]
If you care about speed, use `[array define]` with a pre-allocated max. size. On my laptop, this is twice as fast as your [textfile] solution (18 ms vs. 36 ms)
![sysex](https://user-images.githubusercontent.com/16126632/84606671-409d2280-aea8-11ea-902e-f741fcfb9bab.JPG)
